fjord/suggest/provider_sumosuggest.py

```diff
@@ -0,0 +1,77 @@
+import json
+import logging
+
+import requests
+
+from fjord.suggest import Link, Provider
+
+
+SUMO_SUGGEST_API_URL = 'https://support.mozilla.org/api/2/search/suggest/'
+
+
+logger = logging.getLogger('i.sumosuggest')
+
+
+def get_kb_articles(locale, product, text):
+    """Retrieves kb articles using the SUMO search suggest API
+
+    Throws a variety of exceptions all of which the caller should
+    deal with.
+
+    :arg text: The text to search
+    """
+    params = {
+        'q': text,
+        'locale': 'en-US',
+        'product': 'firefox',
+        'max_documents': 3,
+        'max_questions': 0
+    }
+    headers = {
+        'content-type': 'application/json',
+        'accept': 'application/json',
+    }
+    resp = requests.get(
+        SUMO_SUGGEST_API_URL,
+        data=json.dumps(params),
+        headers=headers,
+        # This is slightly larger than 3 which is what the requests
+        # docs suggest, but we really want the shortest timeout we can
+        # get away with since this happens during the http request
+        # handling cycle.
+        timeout=3.5
+    )
+    return resp.json()['documents']
```

Comment:
What happens when the request times out? I guess it will raise an exception, which will be caught in the provider?